### ISSUE TITLE
Flaps should not work without battery

### ISSUE THREAD

**Issue by ghost** [Harald67/c150#62]
At the moment the flaps are working independent of the electrical system, but as these are electrical flaps, shouldn't they depend on the battery/alternator switched on?

**Comment by gilbertohasnofb** [comment]
No need for the question tag, the flaps are electrical and shouldn't work when the master is off.

**Comment by ghost** [comment]
Okay, that was the question I had, now I can try to do it myself..

**Comment by ghost** [comment]
In the last commit the property /controls/flight/flaps-serviceable will be set to false if no power is there, but flaps are still working, do you know how to change that @gilbertohasnofb @Harald67 ?

**Comment by gilbertohasnofb** [comment]
I don't like using the `serviceable` property for that, I think this should be done via the electrical system. Maybe @Harald67 is the guy to take a look at this.

Also, here is another thing with this type of flap lever: as far as I know, you should hold the lever up or down to bring the flaps up or down while using the scale on the left side of the cockpit to see the position of the flap. That is, there are no notches as in the c172p, so it's up to the pilot to release the lever to the neutral position at the correct time. If that's the case, then we could emulate this behaviour when using the mouse, what do you think @Harald67 ?

**Comment by ghost** [comment]
Yep, IIRC you're right, so the complete flaps lever animation needs to be reversed.

**Comment by Harald67** [comment]
Yes we can do that. So instead of going from one position to another the animation code should just increment the position a bit. The tooltip should already work correctly.

I also agree that serviceable should not be used for that.

**Comment by gilbertohasnofb** [comment]
@Harald67 :+1:
